Default requirements.txt won't work for python3.10

building throws error:
```
  × Running setup.py install for gmpy2 did not run successfully.
  │ exit code: 1
  ╰─> [91 lines of output]
      running install
      /usr/lib/python3/dist-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_ext
      building 'gmpy2' extension
      creating build
      creating build/temp.linux-x86_64-3.10
      creating build/temp.linux-x86_64-3.10/src
      x86_64-linux-gnu-gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DWITHMPFR -DWITHMPC -I/usr/include/python3.10 -c src/gmpy2.c -o build/temp.linux-x86_64-3.10/src/gmpy2.o
      In file included from src/gmpy2.c:426:
      src/gmpy.h:226: warning: "Py_RETURN_NOTIMPLEMENTED" redefined
        226 | #define Py_RETURN_NOTIMPLEMENTED\
            |
      In file included from /usr/include/python3.10/Python.h:74,
                       from src/gmpy2.c:404:
      /usr/include/python3.10/object.h:623: note: this is the location of the previous definition
        623 | #define Py_RETURN_NOTIMPLEMENTED return Py_NewRef(Py_NotImplemented)
            |
      In file included from src/gmpy2.c:509:
      src/mpz_pylong.c: In function ‘mpn_pythonhash’:
      src/mpz_pylong.c:225:19: warning: comparison of integer expressions of different signedness: ‘ssize_t’ {aka ‘long int’} and ‘long unsigned int’ [-Wsign-compare]
        225 |       if (bit_pos <= 8*sizeof(mp_limb_t))
            |                   ^~
      In file included from src/gmpy2.c:517:
      src/gmpy_convert.c: In function ‘Pympc_From_PyStr’:
      src/gmpy_convert.c:2526:20: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘Py_ssize_t’ {aka ‘long int’} [-Wsign-compare]
       2526 |     if (strlen(cp) != len) {
            |                    ^~
      In file included from src/gmpy2.c:529:
      src/gmpy_random.c: In function ‘GMPY_mpfr_grandom’:
      src/gmpy_random.c:251:5: warning: ‘mpfr_grandom’ is deprecated [-Wdeprecated-declarations]
        251 |     mpfr_grandom(Pympfr_AS_MPFR(result1), Pympfr_AS_MPFR(result2),
            |     ^~~~~~~~~~~~
      In file included from src/gmpy.h:252,
                       from src/gmpy2.c:426:
      /usr/include/mpfr.h:535:21: note: declared here
        535 | __MPFR_DECLSPEC int mpfr_grandom (mpfr_ptr, mpfr_ptr, gmp_randstate_t,
            |                     ^~~~~~~~~~~~
      In file included from src/gmpy2.c:541:
      src/gmpy_mpmath.c: In function ‘Pympz_mpmath_normalize’:
      src/gmpy_mpmath.c:103:9: warning: ‘PyUnicode_AsUnicode’ is deprecated [-Wdeprecated-declarations]
        103 |         rnd = Py2or3String_AsString(rndstr)[0];
            |         ^~~
      In file included from /usr/include/python3.10/unicodeobject.h:1046,
                       from /usr/include/python3.10/Python.h:83,
                       from src/gmpy2.c:404:
      /usr/include/python3.10/cpython/unicodeobject.h:580:45: note: declared here
        580 | Py_DEPRECATED(3.3) PyAPI_FUNC(Py_UNICODE *) PyUnicode_AsUnicode(
            |                                             ^~~~~~~~~~~~~~~~~~~
      In file included from src/gmpy2.c:541:
      src/gmpy_mpmath.c:160:53: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘mpir_si’ {aka ‘long int’} [-Wsign-compare]
        160 |                     if (mpz_sizeinbase(lower->z, 2) == shift) {
            |                                                     ^~
      src/gmpy_mpmath.c:162:52: warning: comparison of integer expressions of different signedness: ‘mp_bitcnt_t’ {aka ‘long unsigned int’} and ‘mpir_si’ {aka ‘long int’} [-Wsign-compare]
        162 |                         if (mpz_scan1(lower->z, 0) == shift-1) {
            |                                                    ^~
      src/gmpy_mpmath.c: In function ‘Pympz_mpmath_create’:
      src/gmpy_mpmath.c:246:13: warning: ‘PyUnicode_AsUnicode’ is deprecated [-Wdeprecated-declarations]
        246 |             rnd = Py2or3String_AsString(PyTuple_GET_ITEM(args, 3));
            |             ^~~
      In file included from /usr/include/python3.10/unicodeobject.h:1046,
                       from /usr/include/python3.10/Python.h:83,
                       from src/gmpy2.c:404:
      /usr/include/python3.10/cpython/unicodeobject.h:580:45: note: declared here
        580 | Py_DEPRECATED(3.3) PyAPI_FUNC(Py_UNICODE *) PyUnicode_AsUnicode(
            |                                             ^~~~~~~~~~~~~~~~~~~
      In file included from src/gmpy2.c:541:
      src/gmpy_mpmath.c:314:52: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘mpir_si’ {aka ‘long int’} [-Wsign-compare]
        314 |                     if (mpz_sizeinbase(lower->z, 2)==shift) {
            |                                                    ^~
      src/gmpy_mpmath.c:316:51: warning: comparison of integer expressions of different signedness: ‘mp_bitcnt_t’ {aka ‘long unsigned int’} and ‘mpir_si’ {aka ‘long int’} [-Wsign-compare]
        316 |                         if (mpz_scan1(lower->z, 0)==shift-1) {
            |                                                   ^~
      In file included from src/gmpy2.c:555:
      src/gmpy_mpfr.c: In function ‘_mpfr_hash’:
      src/gmpy_mpfr.c:756:20: error: ‘_PyHASH_NAN’ undeclared (first use in this function); did you mean ‘Py_IS_NAN’?
        756 |             return _PyHASH_NAN;
            |                    ^~~~~~~~~~~
            |                    Py_IS_NAN
      src/gmpy_mpfr.c:756:20: note: each undeclared identifier is reported only once for each function it appears in
      src/gmpy_mpfr.c: In function ‘Pympfr_root’:
      src/gmpy_mpfr.c:988:5: warning: ‘mpfr_root’ is deprecated [-Wdeprecated-declarations]
        988 |     result->rc = mpfr_root(result->f, Pympfr_AS_MPFR(self), n,
            |     ^~~~~~
      In file included from src/gmpy.h:252,
                       from src/gmpy2.c:426:
      /usr/include/mpfr.h:727:21: note: declared here
        727 | __MPFR_DECLSPEC int mpfr_root (mpfr_ptr, mpfr_srcptr, unsigned long,
            |                     ^~~~~~~~~
      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
```
That problem is believed to be a new feature in python3.10: https://bugzilla.redhat.com/show_bug.cgi?id=1959010
And gmpy2==2.0.8 is not ready for this new feature.
gmpy2==2.1.5 (which is the default version at present) is tested and passed the building in python3.10.
I'm not sure if there is any special consideration on gmpy2 version, if not, maybe just update it to be compatible with python3.10.

--- a/README.md
+++ b/README.md
@@ -60,7 +60,7 @@ g++ >= 7.0 or clang >= 10.0
 
 The following libraries and tools are also required,
 ```
-python >= 3.6.9
+Python >= 3.8
 pip >= 22.0.1
 OpenSSL >= 1.1.0
 numa >= 2.0.12

--- a/requirements.txt
+++ b/requirements.txt
@@ -4,6 +4,6 @@
 wheel
 numpy==1.23.1
 pycryptodomex==3.15.0
-gmpy2==2.0.8
+gmpy2==2.1.5
 cachetools==3.0.0
 ruamel-yaml==0.16.10

--- a/setup.py
+++ b/setup.py
@@ -86,7 +86,7 @@ setup(
         "wheel",
         "numpy==1.23.1",
         "pycryptodomex==3.15.0",
-        "gmpy2==2.0.8",
+        "gmpy2==2.1.5",
         "cachetools==3.0.0",
         "ruamel.yaml==0.16.10",
     ],

